hsc/crawler.py

```diff
@@ -149,27 +80,69 @@ def store_submission(self, file_name, code):
 		with open(file_name, 'w') as text_file:
 			text_file.write(code)
 
-	def update_readme(self, readme_file_path, problem_readme_text):
+	def update_readme(self, readme_file_path, readme_text):
 		header_length = self.readme_headers_len
 		with open(readme_file_path, 'r+') as text_file:
 			lines = text_file.readlines()
-			lines.append(problem_readme_text)
+			lines.append(readme_text)
 			sortedlines = lines[:header_length] + sorted(lines[header_length:])
 			text_file.seek(0)
 			text_file.writelines(sortedlines)
 
-	def create_readme(self, track_name, track_url, file_name):
-		if track_name is not None:
-			os.makedirs(os.path.dirname(file_name), exist_ok=True)
-			text = self.new_readme_text.format(track_name, track_url)
-			with open(file_name, 'w') as text_file:
-				text_file.write(text)
-
-	def get_file_path(self, folder_name, file_name_with_extension):
-		return os.path.join(self.base_folder_name, folder_name, file_name_with_extension)
+	def write(self, file_name, text):
+		os.makedirs(os.path.dirname(file_name), exist_ok=True)
+		with open(file_name, 'w') as text_file:
+			text_file.write(text)
+
+
+	def create_readmes(self, domain_name, subdomain_name, domain_url, subdomain_url, subdomain_readme_path, domain_readme_path, root_readme_path):
```

Comment:
too many arguments.. can it be reduced? 
If not at least consider breaking this line into multiple

Reply:
Fixed in 9e7e65c